### PR TITLE
Add back the stdoutmetric exporter

### LINKED ISSUE #2805: Add back the stdoutmetric exporter code
Blocked by the metric SDK having a working example in the new_sdk/main branch

Removed in https://github.com/open-telemetry/opentelemetry-go/pull/2802

### PR DESCRIPTION
**Pull request by MrAlias**
Work in progress

Resolves #2805

This uses an added `Encoder` interface to allow output customization and extensibility of the exporter. It continues to provide the default JSON output.

### CODE REVIEW

File: sdk/metric/metricdata/temporality.go [hunk 1]
```diff
@@ -35,3 +35,18 @@ const (
...
+
+var temporalityString = map[Temporality]string{
+	CumulativeTemporality: "cumulative",
+	DeltaTemporality:      "delta",
+}
+
+// String returns the string name of t.
+func (t Temporality) String() string {
+	return temporalityString[t]
+}
```
Comment (MadVikingGod) [inline sdk/metric/metricdata/temporality.go:47]: Do we want to use `striger` for this?

File: exporters/stdout/stdoutmetric/config.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	"os"
+)
+
+// config contains options for the exporter.
+type config struct {
+	encoder *encoderHolder
+}
+
+// newConfig creates a validated config configured with options.
+func newConfig(options ...Option) (config, error) {
```
Comment (MadVikingGod) [inline exporters/stdout/stdoutmetric/config.go:31]: This function doesn't have a way to return errors.  Can we drop the `error`?

**Review by MadVikingGod — COMMENTED** [review]
This looks good.  I will close #3056 in favor of this.

I took the shutdown tests from `stdouttrace`, if you want a quick set of tests for that functionality.

**Comment by jmacd** [comment]
I've become used to reading the OTel Collector's logging exporter's output. That would be another nice alternative, here.

**Review by jmacd — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Pull request merged**